### PR TITLE
GPS tab: New design for gps Status / Quality

### PR DESCRIPTION
**Pull request by HThuren**
New design for gps Status / Quality

1st column - Status:    unused = 'depends', USED = green
2nd column - Quality: no signal = red, unusable = red, searching = red, locked = orange and fully locked = green
'depends' = if not USED, then colour same as Quality

image

### CODE REVIEW

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> FAIL
 - cooling off period lapsed -> FAIL
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> FAIL
 - `Tested` label found -> FAIL
 - assigned to an approver -> FAIL
 - approver count at least three -> FAIL

**Comment by ctzsnooze** [comment]
This sounds really good!  I look forward to testing.  Thank you!

**Comment by HThuren** [comment]
By the way, Pls feedback on colour Yellow, rgb yellow was very light, so I use more orange tones, but then move against rgb orange.

**Comment by nerdCopter** [comment]
> By the way, Pls feedback on colour Yellow, rgb yellow was very light, so I use more orange tones, but then move against rgb orange.

can you send screenshot.  recommend re-using any existing colors that are already used in current CSS files.

**Review by nerdCopter — APPROVED** [review]

File: src/js/tabs/gps.js
Comment (chmelevskij) [inline src/js/tabs/gps.js:277]:
```suggestion
```

**Comment by ctzsnooze** [comment]
The colours are really nice!
Some thoughts....
I guess we want to see 'green lights' when things are good.

The most important word is 'used'.  It means that the satellite is actively involved in calculating the fix, based on signal strength and position.

That's probably the one we want to see standing out by how 'green' it is.  It should probably be the leftmost icon in the row.

Healthy refers to signal strength, and when unhealthy, will not be used.  'searching' essentially is a form of 'unhealthy', it means the signal had been received previously, but not now.With either of these flags, the satellite isn't used.  

Hence I think the hierarchy goes used > fully locked > locked > healthy > searching > unhealthy

I wonder how it would look with only 'Used' in green, perhaps with the button width a bit wider, and in capitals?  Then fully locked and locked could be the orange colours. For signal, healthy could be the same orange as locked, and searching / unhealthy could be grey.

**Comment by haslinghuis** [comment]
Agree. Used is most important here. Asked [be]for[e] all three coloms to use green before when in optimal state.

**Comment by HThuren** [comment]
Ok, now we have this design.
1st column: unused = red, used = green
2nd column: non healthy = red, healthy = green
3d column: unusable = red, searching = yellow, locked = orange and fully locked = green

what do you think ?

image

**Comment by ctzsnooze** [comment]
Thank you!  I woke up early before going to work, and thought it looked more 'obvious' in terms of explaining which were the satellites we were actually using for our 3D fix.  The green 'USED' signs were very clear.

I found this information about signal quality:
```
Signal quality indicator:
• 0 = no signal
• 1=searchingsignal
• 2=signalacquired
• 3=signaldetectedbutunusable
• 4=codelockedandtimesynchronized
• 5,6,7=codeandcarrierlockedandtime
synchronized
```
And signal health flag
```
Signal health flag:
• 0=unknown
• 1=healthy
• 2=unhealthy
```

**Comment by ctzsnooze** [comment]
I have a feeling that whenever we get unhealthy or unknown, we get red on 'unused', but have to check.

**Comment by HThuren** [comment]
> I have a feeling that whenever we get unhealthy or unknown, we get red on 'unused', but have to check.

A another design are to show "unused" in grey and green USED, and omit the many red boxes.

**Comment by HThuren** [comment]
Give this:
image
image
image
image
image

**Comment by ctzsnooze** [comment]
I guess it would be a 'community' decision, but I wonder about the need for the 'healthy' column ?

**Comment by HThuren** [comment]
> I guess it would be a 'community' decision, but I wonder about the need for the 'healthy' column ?

You may have a point, healty came up before anything else, but USED seems to be influenced by quality indicator

**Comment by ctzsnooze** [comment]
PS any chance we could show lat and long with 6 decimal places?  The last two decimal places can be useful as an indicator of stability.

**Comment by HThuren** [comment]
> PS any chance we could show lat and long with 6 decimal places? The last two decimal places can be useful as an indicator of position stability.

Yes, no issue, changed to fixed at 6

**Comment by ctzsnooze** [comment]
Thank you!  Unfortunately when I make the latest build, the yellow and grey colours have disappeared, it's only red and green.  
Screen Shot 2023-09-09 at 22 21 06

**Comment by ctzsnooze** [comment]
Thanks, again!
The number of decimal places for lat and lon are still showing 4, perhaps it needs changing in other places also?
I was able to try some options and ended up with:
- replaced 'grey' colour with 'red'
- removed the 'healthy/unhealthy' column
- if the satellite was not 'in used', then I set the 'not used' button to the colour of the 'signal' column
It ended up like this:
Screen Shot 2023-09-09 at 23 29 16

**Comment by HThuren** [comment]
regarding 6 digit, not yet uploaded :-)

**Comment by HThuren** [comment]
Here we go
image

**Comment by haslinghuis** [comment]
Looking good! Please rebase and resolve sonarcloud issues.

**Comment by haslinghuis** [comment]
5 code smells remaining :yum:

File: src/js/tabs/gps.js
Comment (haslinghuis) [inline src/js/tabs/gps.js:276]: Fix your code by using else if if the nested if is the only statement in the else block.

Comment (HThuren) [reply src/js/tabs/gps.js]: on it way :-)

**Comment by ctzsnooze** [comment]
@HThuren thank you!  That looks absolutely great!  Very easy to interpret now.  Approved!

Also the 6 decimal places is quite useful.  If only the last digit is changing, the reading is probably inside a 1-2m circle.  If the 5th digit is changing a lot, then we have low accuracy.

If we had access to the 'dilution of precision value' (a multiple of the specified accuracy of the module), or its newer and more precise horizontal accuracy (hAcc) value (a value in metres), it would be awesome if, on the map, we drew a grey circle around the current point, the radius of which could indicate current precision.
We could do something similar from the reported speed value.  A smoothed average of the speed value, with say a time constant of half a second, would, while the quad was still, relate to accuracy.  And we already have the speed value.

A related concept could be used with our 'distance to home' value.  I had wondered if we could have a button which 'set' the home point to the current latitude and longitude.  The user could click it, then get a distance traveled from home indicator.  Once that distance stabilised a bit, they could click again, and watch for absolute drift over time with that specific GPS module.

Both these tools would be quite useful for assessing the accuracy of the GPS and monitoring how that evolved over time.

**Review by ctzsnooze — APPROVED** [review]

**Review by haslinghuis — APPROVED** [review]

File: locales/en/messages.json
Comment (McGiverGim) [inline locales/en/messages.json:2808]: Sorry, I arrive too late to this, I was out... this is not the way to do that. Using "spaces" to align elements will break the alignment with the translations.

Comment (HThuren) [reply locales/en/messages.json:2808]: you are right, this was forgotten in the battle.
I think I change in a new PR, not to confuse with this one

**Pull request merged**
